src/Serializers/FingerprintSerializer.php

```diff
@@ -48,15 +47,11 @@ public function addIdToSerialization( Entity $entity, array &$serialization ) {
 		$serialization['id'] = $id->getSerialization();
 	}
 
-	public function addLabelsToSerialization( Entity $entity, array &$serialization ) {
-		$labels = $entity->getFingerprint()->getLabels();
-
+	public function addLabelsToSerialization( TermList $labels, array &$serialization ) {
```

Comment:
Many methods here can be private. Not part of this patch, just wanted to point this out.